test/bootc/utils.sh

```diff
@@ -0,0 +1,284 @@
+repo_gpgcheck=0
+enabled=1
+enabled_metadata=1
+REPO_EOF
+RUN dnf install -y go-fdo-client
+EOF
+  fi
+
+  podman build --retry=5 --retry-delay=10s -t "fdo-bootc:latest" -f Containerfile .
+}
+
+generate_iso_from_bootc() {
+  local bootc_image_builder=$1
+  rm -fr output
+  mkdir -pv output
+  sudo podman run \
+      --rm \
+      -it \
+      --privileged \
+      --pull=newer \
+      --security-opt label=type:unconfined_t \
+      -v "$(pwd)/output:/output" \
+      -v "/var/lib/containers/storage:/var/lib/containers/storage" \
+      "${bootc_image_builder}" \
+      --type anaconda-iso \
+      --rootfs xfs \
+      --use-librepo=true \
+      "localhost/fdo-bootc:latest"
+}
+
+generate_kickstart_iso() {
+    rm -fr /var/lib/libvirt/images/install.iso
+    isomount=$(mktemp -d)
+    sudo mount -v -o "ro" "output/bootiso/install.iso" "$isomount"
+    new_ks_file="bib.ks"
+    cat > "${new_ks_file}" << EOFKS
+text
+$(cat "${isomount}/osbuild-base.ks")
+$(cat "${isomount}/osbuild.ks")
+EOFKS
+    sed -i '/%include/d' "${new_ks_file}"
+    sed -i '/%post --erroronfail/i\
+user --name=admin --groups=wheel --homedir=/home/admin --iscrypted --password=\$6\$GRmb7S0p8vsYmXzH\$o0E020S.9JQGaHkszoog4ha4AQVs3sk8q0DvLjSMxoxHBKnB2FBXGQ/OkwZQfW/76ktHd0NX5nls2LPxPuUdl.' "${new_ks_file}"
+    sed -i "/%post --erroronfail/i\
+sshkey --username admin \"${ssh_key_pub}\"" "${new_ks_file}"
+    sed -i "/bootc switch/a\
+go-fdo-client --blob /boot/device_credential --debug device-init http://${manufacturer_ip}:8038 --device-info=iot-device --key ec256" "${new_ks_file}"
+    sed -i '/bootc switch/a\
+echo "admin ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/admin' "${new_ks_file}"
+    log_info "==== New kickstart file ===="
+    cat "${new_ks_file}"
+    log_info "============================"
+    log_info "Writing new ISO"
+    sudo mkksiso -c "console=ttyS0,115200" --ks "$new_ks_file" "output/bootiso/install.iso" "/var/lib/libvirt/images/install.iso"
+    sudo umount -v "$isomount"
+    rm -rf "$isomount"
+}
+
+install_client() {
+  local base_image=$1
+  local bootc_image_builder=$2
+
+  # Install required package and configure libvirt
+  configure_libvirt
+
+  # Configure firewall
+  configure_firewall
+
+  # Build container and generate ISO
+  log_info "Building bootc container with go-fdo-client installed"
+  build_bootc_container $base_image
+
+  log_info "Using bootc image builder to generate anaconda-iso"
+  generate_iso_from_bootc $bootc_image_builder
+
+  # Create modified kickstart file
+  log_info "Generating kickstart file and mkksiso"
+  generate_kickstart_iso
+}
+
+configure_firewall() {
+    systemctl start firewalld
+}
+
+configure_libvirt() {
+    log_info "Install required packages"
+    local packages=(golang make podman jq qemu-img qemu-kvm libvirt-client libvirt-daemon-kvm libvirt-daemon virt-install ansible-core firewalld lorax gobject-introspection)
+    dnf install -y "${packages[@]}"
```

Comment:
I would move the installation of each package to the function that needs it and they need to be installed if not running on tmt: if the test is running in tmt the packages must be installed in the prepare section so they are available to all the tests.